lib/mayaUsd/fileio/shading/shadingModeUseRegistry.cpp

```diff
@@ -180,7 +185,7 @@ class UseRegistryShadingModeExporter : public UsdMayaShadingModeExporter
             rootPlugCopy,
             MFn::kInvalid,
             MItDependencyGraph::Direction::kUpstream,
-            MItDependencyGraph::Traversal::kDepthFirst,
```

Comment:
I don't understand in which way switching to kBreadthFirst helps here and what it solves.

Reply:
Lets have this scenario:
![image](https://github.com/Autodesk/maya-usd/assets/56274617/6d3726ab-14a7-4046-8af3-ec7a9993e486)
When traversing depth first the connections will be returned in this order:
```
file1.outColor -> standardSurface2.base_color <== Exports file1
file3.outColor -> file1.colorGain <== Exports file3
file3.outColor -> file2.colorGain <== Incorrect since file2 is not yet exported
```
Traversing breadth first will produce this order:
```
file1.outColor -> standardSurface2.base_color <== Exports file1
file2.outColor -> standardSurface2.specular_color <== Exports file2
file3.outColor -> file1.colorGain <== Exports file3
file3.outColor -> file2.colorGain <== Correct now since we have already exported file2
```

Reply:
In my mind, we should not make a move file3.outColor -> file2.colorGain because this traverses the graph in the opposite direction. Since file2 is connected is connected to standardSurface2, it will be processed in its turn

Reply:
Here is the order of the connections returned by a breadth first MItDependencyGraph:
![Graph1_Breadth_First](https://github.com/Autodesk/maya-usd/assets/56274617/ee6eaee4-7f47-4a5c-87af-4eeb4e90b078)
As you can see we end up traversing "file3.outColor -> file2.colorGain" at step 5. Since we already processed "file2.outColor -> standardSurface.pecularColor" at step 2 then we are OK.
Still in full agreement that browsing all the destinations of a source port is bad. I think removing that part would indeed work perfectly fine since the traversal will eventually hit all connections.